overlord/snapstate/reboot.go

```diff
@@ -115,6 +115,14 @@ func deviceModelBootBase(st *state.State, providedDeviceCtx DeviceContext) (stri
 	return bootBase, nil
 }
 
+func bootBaseSnapType(byTypeTss map[snap.Type]*state.TaskSet, bootBase string) snap.Type {
```

Comment:
super nitpick, `bootBase` here doesn't seem to be used, maybe we can remove it?

Reply:
This is a minor nitpick, please feel free to ignore it.

Actually after second thought, it would be more explicit if we could check `byTypeTss[snap.TypeBase]` snap name equals `bootBase`. My concern is that it depends on the implementation details of `taskSetsByTypeForEssentialSnaps` that it only sets `byTypeTss[snap.TypeBase]` when it finds `bootBase`.

Reply:
Super good catch with the `bootBase`, it's removable.

While I get what you mean, it's a helper function to extract the right type out of `byTypeTss`, and not a function meant to stand on it's own, it's directly coupled to `byTypeTss`, and thus we don't need to take into account that some day we want to change what `byTypeTss` means, because then logic surrounding `byTypeTss` most certainly needs to change anyway.

Reply:
makes sense, thanks for the explanation!